src/enums/ClaimConditionType.ts

```diff
@@ -0,0 +1,27 @@
+enum ClaimConditionType {
+  StartTime,
+  LimitPerWallet,
+  MerkleRoot,
+  WaitTimeBetweenClaims,
+}
+
+/**
+ * Questions:
+ *
+ * 1. Can there only be a single `startTimestamp` condition? What is a `startTimestamp` condition?
+ *
+ * 2. What is `maxMintSupply`?
+ *
+ * 3. What is `currentMintSupply`?
+ *
+ * 4. What is `quantityLimitPerTransaction`?
+ *
+ * 5. What is `waitTimeSecondsLimitPerTransaction`?
```

Comment:
how long do you have to wait before you can `claim` again.

so if you set `waitTimeSecondsLimitPerTransaction` to MAX_UINT256
you can only `claim` ONCE because you're probably not going to be alive to claim the next one.

if you set it to 15 seconds, you can `claim` every 15 seconds.

if you combine this to MAX_UINT and `quantityLimitPerTransaction` to 1. you'll only be able to claim 1 maximum. in your lifetime at least.

Reply:
> you can only claim ONCE because you're probably not going to be alive to claim the next one.

💀

This makes sense. Composing these conditions is going to be very fun 🤩

Reply:
yeah, i think we basically want to simplify that in the sense that we'd have a method that does:
max claim quantity per wallet or etc. abstract that away and make it presentable for dev / user.